7_react/react1/week2/hw-react-week2/src/components/TaskList.jsx

```diff
@@ -0,0 +1,37 @@
+import Checkbox from "./Checkbox";
+
+const TaskList = props => {
+  const { list, setList } = props;
+  const onChangeStatus = element => {
+    const { name, checked } = element.target;
+
+    const updateList = list.map(item => ({
+      ...item,
+      done: item.id === name ? checked : item.done
+    }));
+    setList(updateList);
+  };
+
+  const onClickRemoveItem = () => {
+    const updateList = list.filter(item => !item.done);
+    setList(updateList);
+  };
+
+  const checkList = list.map(item => (
+    <Checkbox key={item.id} data={item} onChange={onChangeStatus} />
+  ));
+  return (
+    <div className="todoList">
+      {list.length ? checkList : "Add your tasks"}
+      {list.length ? (
+        <p>
+          <button className="DeleteButtom" onClick={onClickRemoveItem}>
+           {'Remove the selected task(s)'}
+          </button>
+        </p>
+      ) : null}
+    </div>
+  );
+};
```

Comment:
It is good that you challenged yourself and made your UI more advanced that what the homework required.  💪 

In the real world you have to be careful about freestyling too much, or at least make sure that you present your ideas with your manager or customer and make sure they agree on your ideas. Otherwise, it may bite you in the 🍑. 

Lets pretend I made a long Todo list for the entire month. I have more than 100 different todos and I am slowly making by way through the list and checking off a few todos per day. It really motivates me to see my progress. However, my nephew got hold of my tablet and added 5 todos of `play video games`. I want to remove these 5 todos but I don't want to lose my history of all the items I have finished so far. What should I do?

If this was not properly aligned with your manager, the error would fall on you unfortunately 😢